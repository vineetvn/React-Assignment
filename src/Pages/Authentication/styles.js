import styled from 'styled-components'
import Input from '../../Common Components/Input/Input'

export const AuthWrapper = styled.div `
    display: flex;
    height: auto;

    @media only screen and (max-width: 768px) {
        flex-direction: column;
    }
`

export const Image = styled.div `
    flex: 60%;
    height: auto;

    @media only screen and (max-width: 768px) {
        height: 400px;
    }
`
export const Img = styled.img `
    width: 100%;
    height: 100%;
`

export const Form = styled.div `
    flex: 40%;
    margin: 1em;
    box-sizing: border-box;
    font-family: ${props => props.theme.fontFamily.Lato};
`

export const StyledInput = styled(Input) `
    border-bottom: 1px solid black;
`
export const Label = styled.div `
    color: ${props => props.theme.color.footerColor};
    font-size: ${props => props.theme.fontSize.productName};
    line-height: ${props => props.theme.lineHeight.auth};
    font-family: ${props => props.theme.fontFamily.Montserrat};
    font-weight: ${props => props.theme.fontWeight.bold};
    padding: 15px 0 15px 20px;
`

export const Div = styled.div `
    margin-top: 50px;
    display: flex;
    justify-content: space-around;
`
export const ShowPassword = styled.div `
    float: right;
    color: ${props => props.theme.backgroundColor.green};

`

export const Button = styled.button `
    outline: none;
    border: none;
    padding: 20px 50px;
    border-top-left-radius: 50px;
    border-bottom-left-radius: 50px;
    border-bottom-right-radius: 50px;
    border-top-right-radius: 50px;
    background-color: ${props => props.theme.backgroundColor.green};

    @media only screen and (max-width: 500px) {
        padding: 20px;
    }
`

export const A = styled.div `
    padding: 20px 50px;
    border-radius: 30px;
    background-color: ${props => props.theme.backgroundColor.lightGrey};
`