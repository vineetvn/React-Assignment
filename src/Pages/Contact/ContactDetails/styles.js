import { Link } from 'react-router-dom';
import styled from 'styled-components';

const Div = styled.div `
    flex: 33.34%;
    color: ${props => (props.theme.color.lightGray)};
    width: 100%;
`
const H3 = styled.h3 `
    color: ${props => (props.theme.color.darkCharcoal)};
    font-size: ${props => (props.theme.fontSize.large)};
    line-height: ${props => (props.theme.lineHeight.xlarge)};
    letter-spacing: ${props => (props.theme.letterSpacing.xsmall)};
    font-weight: ${props => (props.theme.fontWeight.medium)};
    font-family: ${props => (props.theme.fontFamily.Montserrat)};
`

const P = styled.p `
    font-size: ${props => (props.theme.fontSize.small)};
    line-height: ${props => (props.theme.lineHeight.large)};
    font-weight: ${props => (props.theme.fontWeight.regular)};
    font-family: ${props => (props.theme.fontFamily.Lato)};
    padding-right: 50px;

    @media only screen and (max-width: 900px) {
        padding-right: 0;
    }
`

const Ul = styled.ul `
    padding: 0;
    list-style: none;
    font-size: ${props => (props.theme.fontSize.small)};
    line-height: ${props => (props.theme.lineHeight.large)};
    font-weight: ${props => (props.theme.fontWeight.regular)};
    font-family: ${props => (props.theme.fontFamily.Lato)};
`

const Li = styled.li `
    display: flex;
    padding: 10px 10px 10px 0;
    display: flex;
    align-items: center;
`

const Span = styled.span `
    margin-right: 30px;
    font-size: ${props => (props.theme.fontSize.large)};
`

const A = styled(Link) `
    text-decoration: none;
    color: ${props => (props.color)};
    &:hover {
        color: ${props => (props.theme.color.lightGray)};
    }
`
export {Div, H3, P, Ul, Li, Span, A};