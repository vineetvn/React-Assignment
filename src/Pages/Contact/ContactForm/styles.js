
import styled from 'styled-components';
import InputElement from '../../../Common Components/Input/Input';

const Div = styled.div `
    flex: 66.66%;
`
const H3 = styled.h3 `
    color: ${props => (props.theme.color.darkCharcoal)};
    font-size: ${props => (props.theme.fontSize.large)};
    line-height: ${props => (props.theme.lineHeight.xlarge)};
    letter-spacing: ${props => (props.theme.letterSpacing.xsmall)};
    font-weight: ${props => (props.theme.fontWeight.medium)};
    font-family: ${props => (props.theme.fontFamily.Montserrat)};
    margin-left: 20px;

    @media only screen and (max-width: 900px) {
        margin-left: 0
    }
`
const Div2 = styled.div `
    width: 100%;
    display: flex;
    flex: 50%;
    box-sizing: border-box;

    @media only screen and (max-width: 900px) {
        flex-direction: column;
    }
`
const Div3 = styled.div `
    width: 30%;
    margin: 0 20px 20px 20px;
    box-sizing: border-box;
`

const Input = styled(InputElement)`
    color: ${props => (props.theme.color.gray)};
    font-size: ${props => (props.theme.fontSize.xsmall)};
    line-height: ${props => (props.theme.lineHeight.medium)};
    font-weight: ${props => (props.theme.fontWeight.medium)};
    font-family: ${props => (props.theme.fontFamily.Lato)};
`

const StyledButton = styled.button `
    text-decoration: none;
    color: ${props => (props.color)};
    background-color: ${props => (props.theme.backgroundColor.white)};
    font-size: ${props => (props.theme.fontSize.xxxsmall)};
    line-height: ${props => (props.theme.lineHeight.medium)};
    letter-spacing: ${props => (props.theme.letterSpacing.small)};
    font-weight: ${props => (props.theme.fontWeight.mediumBold)};
    font-family: ${props => (props.theme.fontFamily.Montserrat)};
    border: 2px solid ${props => (props.color)};
    padding: 12px 24px;
    white-space: nowrap;
    border-bottom-left-radius: 50px;
    border-top-left-radius: 50px;
    border-bottom-right-radius: 50px;
    border-top-right-radius: 50px;

    @media only screen and (max-width: 500px) {
        letter-spacing: 0;
    }
`

export {Div, H3, Div2, Div3, Input, StyledButton};