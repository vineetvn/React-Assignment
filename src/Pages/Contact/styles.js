import styled from 'styled-components';
import OfficeImg from '../../Assets/Images/our office.jpg'
export const Div = styled.div `
    display: flex;
    margin: 50px 100px;

    @media only screen and (max-width: 900px) {
        flex-direction: column;
        margin: 50px 70px;;
    }
`

export const Location = styled.div `
    display: flex;
    width: 100%;
    height: 300px;

    @media only screen and (max-width: 900px) {
        flex-direction: column;
        height: 500px;
    }
`

export const Office = styled.div `
    background: linear-gradient(rgba(0,0,0,0.8), rgba(0,0,0,0.8)), url('${OfficeImg}');
    background-position: center;
    flex: 30%;
    height: 100%;
    text-align: center;
    display: flex;
    flex-direction: column;
    align-content: center;
    justify-content: center;

    @media only screen and (max-width: 900px) {
        flex: 50%;
    }
`

export const LocImage = styled.img `
    flex: 70%;
    height: 100%;
    object-fit: cover;
    overflow: hidden;

    @media only screen and (max-width: 900px) {
        flex: 50%;
    }
`

export const H3 = styled.h3 `
    font-family: ${props => props.theme.fontFamily.Montserrat};
    font-size: ${props => props.theme.fontSize.large};
    line-height: ${props => props.theme.lineHeight.xlarge};
    color: ${props => props.theme.color.white};
    font-weight: ${props => props.theme.fontWeight.bold};
`

export const Li = styled.li `
    list-style: none;
    font-family: ${props => props.theme.fontFamily.Lato};
    font-size: ${props => props.theme.fontSize.productName};
    line-height: ${props => props.theme.lineHeight.xlarge};
    color: ${props => props.theme.color.white};
    font-weight: ${props => props.theme.fontWeight.regular};
    text-align: center;
`