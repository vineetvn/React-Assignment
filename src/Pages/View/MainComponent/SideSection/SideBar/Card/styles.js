import styled from 'styled-components';

export const Div = styled.div `
    display: flex;
    padding: 20px 0;
    border-bottom: 1px solid rgba(0,0,0,0.2)
`
export const Img = styled.img `
    flex: 25%;
    width: 100px;
`
export const Details = styled.div `
    flex: 75%;
    padding-left: 20px;
    font-family: ${props => props.theme.fontFamily.Montserrat};
`