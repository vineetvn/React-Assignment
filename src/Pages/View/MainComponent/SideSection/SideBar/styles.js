import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const Div = styled.div `
    flex: 25%;
    margin: 50px 10px;
    font-family: ${props => props.theme.fontFamily.Montserrat};

    @media only screen and (max-width: 500px) {
        margin: 30px;
    }
`

export const H3 = styled.h3 `
    font-family: ${props => props.theme.fontFamily.Montserrat};
`