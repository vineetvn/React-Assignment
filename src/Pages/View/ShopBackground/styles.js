import styled from 'styled-components';

import background from '../../../Assets/Images/shopImage.jpg';

export const Div = styled.div `
    background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url('${background}');
    background-size: cover;
    background-position: center;
    position: relative;
    height: 500px;
`

export const ImageHeading = styled.div `
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    color: ${props => props.theme.color.white};
    font-size: ${props => props.theme.fontSize.large};
    font-family: ${props => props.theme.fontFamily.Montserrat};
    text-align: center;
`