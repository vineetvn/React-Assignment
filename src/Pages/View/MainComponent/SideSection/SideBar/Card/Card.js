import React, { Component } from 'react'
import { Details, Div, Img } from './styles'

export default class Card extends Component {
    constructor(props) {
        super(props)
    
        this.state = {
             clicked: false
        }
    }
    
    clickHandler() {
        this.setState({clicked: true})
    }
    render() {
        return (
            <Div>
                <Img src= {`../../../../../../Assets/Images/${this.props.img}`}></Img>
                <Details>
                    <h4>{this.props.name}</h4>
                    <p>$ {this.props.price}</p>
                </Details>
            </Div>
        )
    }
}
