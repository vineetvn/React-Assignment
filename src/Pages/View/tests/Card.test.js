import Adapter from 'enzyme-adapter-react-16';
import {configure} from 'enzyme';
import { shallowRender } from '../../../testUtils';
import Card from '../MainComponent/SideSection/SideBar/Card/Card';
import { Details, Img } from '../MainComponent/SideSection/SideBar/Card/styles';

configure({adapter: new Adapter()});

describe('SideBar Card', () => {
    let wrapper;

    let props = {
        img: 'Backpack.jpg',
        name: 'BackPack',
        price: '$10'
    }

    beforeEach(() => {
        wrapper = shallowRender(Card, false, props);
    })

    it('should render component', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('should find img component', () => {
        expect(wrapper.find(Img))
    });

    it('should render details of the card', () => {
        expect(wrapper.find(Details))
    });
});