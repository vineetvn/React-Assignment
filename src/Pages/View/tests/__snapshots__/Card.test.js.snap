// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Card should render component 1`] = `ShallowWrapper {}`;
